src/averageAges.js

```diff
@@ -59,8 +77,41 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const childrens = people.filter(
+    person =>
+      people.some(mother => mother.name === person.mother)
+        && (onlyWithSon
+          ? isMale(person)
+          : true)
+  );
+
+  return childrens.reduce((sum, child) => {
+    const mother = people.find(
+      person =>
+        child.mother === person.name
+    );
+
+    return sum + (child.born - mother.born);
+  }, 0) / childrens.length;
```

Comment:
Don't forget to use destructuring for objects.
```suggestion
  const ageDiff = children.reduce((sum, { born, mother: motherName }) => {
    const motherBirthYear = people.find(({ name })=> name === motherName).born;
    return sum + (born - motherBirthYear);
  }, 0);
  return ageDiff / children.length;
```